cdragontoolbox/sknfile.py

```diff
@@ -0,0 +1,71 @@
+from .tools import BinParser
+
+
+class SKNFile:
+    def __init__(self, file):
+        if isinstance(file, str):
+            file = open(file, "rb")
+        if file.read(4) != b"\x33\x22\x11\x00":
+            raise ValueError("missing magic code")
+
+        f = BinParser(file)
+
+        self.major, self.minor, self.count = f.unpack("<HHI")
```

Comment:
`count` is just a temporary and does not need to be stored on `self`.
If you need the entry count afterwards, you can already use `len(self.entries)`.